### PR TITLE
Expose max_muxing_queue_size to user

### PR DESCRIPTION
**Pull request by nyanmisaka**
Part1: this
Part2: https://github.com/jellyfin/jellyfin-web/pull/1842

**Changes**
- Expose max_muxing_queue_size to user. see https://github.com/jellyfin/jellyfin/pull/3975

### CODE REVIEW

File: Jellyfin.Api/Controllers/DynamicHlsController.cs [hunk 1]
```diff
@@ -1354,15 +1354,20 @@ private string GetCommandLineArguments(string outputPath, EncodingOptions encodi
                 segmentFormat = "mpegts";
             }
 
+            var maxMuxingQueueSize = encodingOptions.MaxMuxingQueueSize >= 128 && encodingOptions.MaxMuxingQueueSize <= int.MaxValue
```
Comment (Bond-009) [inline Jellyfin.Api/Controllers/DynamicHlsController.cs:1357]: Why `encodingOptions.MaxMuxingQueueSize <= int.MaxValue`?

Comment (nyanmisaka) [reply Jellyfin.Api/Controllers/DynamicHlsController.cs:1357]: `The value for max_muxing_queue_size was 9999999999999999 which is not within -2147483648.000000 - 2147483647.000000`

FFmpeg only accepts a reasonable int32 value for max_muxing_queue_size .

Comment (Bond-009) [reply Jellyfin.Api/Controllers/DynamicHlsController.cs:1357]: But `MaxMuxingQueueSize` is an int32, so how would it hold a larger value?

Comment (nyanmisaka) [reply Jellyfin.Api/Controllers/DynamicHlsController.cs]: Well. This check is redundant.

File: Jellyfin.Api/Controllers/DynamicHlsController.cs [hunk 2]
```diff
@@ -1354,15 +1354,20 @@ private string GetCommandLineArguments(string outputPath, EncodingOptions encodi
                 segmentFormat = "mpegts";
             }
 
+            var maxMuxingQueueSize = encodingOptions.MaxMuxingQueueSize >= 128
```
Comment (Bond-009) [inline Jellyfin.Api/Controllers/DynamicHlsController.cs:1357]:
```suggestion
            var maxMuxingQueueSize = encodingOptions.MaxMuxingQueueSize > 128
```
Minor nitpick, but this way we don't need to convert an int to a string when the value is 128

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**